deploy/charts/cert-manager/templates/cainjector-deployment.yaml

```diff
@@ -16,7 +16,7 @@ metadata:
   {{- end }}
 spec:
   replicas: {{ .Values.cainjector.replicaCount }}
-  {{- if ne (quote .Values.global.revisionHistoryLimit) (quote "") }}
+  {{- if ne (trimAll "'" (squote .Values.global.revisionHistoryLimit)) "" }}
   revisionHistoryLimit: {{ .Values.global.revisionHistoryLimit }}
   {{- end }}
```

Comment:
Elsewhere in this chart we use `if hasKey .Values.global "revisionHistoryLimit`:

https://github.com/cert-manager/cert-manager/blob/6a9c402c61ccb1b85c8c0f44d16a3ac2d91c61f0/deploy/charts/cert-manager/templates/webhook-deployment.yaml#L47-L49

Would that technique work here?

Reply:
@wallrj we also want to skip the property when the value is not set or set to an empty value in the values.yaml file or using a command line flag. This solution would also set the property when the value is just an empty string.

Reply:
If the user has used `--set global.revisionHistoryLimit=` then `revisionHistoryLimit: ` should appear in the rendered manifests, since that is what they have asked for.